### ISSUE TITLE
[Feature request]: Allow using multiple base apps

### ISSUE THREAD

**Issue by TheEvilSkeleton** [flatpak/flatpak#4797]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a feature request that matches the one I want to file, without success.

### Suggestion

Currently, we can only use one base app. While this is fine for the most part, some apps would want to have two or more base apps. At the moment we're in this awkward situation where we need to use one base app and build all the dependencies of another base app, which contribute to duplicating work.

In my opinion, supporting multiple base apps would make it much easier to maintain applications, and also avoid duplicating work as much as possible.

Some real life examples:
1. Heroic Games Launcher is an Electron application for Epic Games Store. It has the feature of optionally using the system Wine. However, we cannot use the Electron base app with the Wine base app, so we have to build dependencies manually, like zypak, and use the Wine base app. https://github.com/flathub/flathub/pull/2897#issuecomment-1061443211
2. Likewise, Fightcade is an Electron application with support for retro games. It uses Wine itself. However, the same issue arises as 1. https://github.com/flathub/com.fightcade.Fightcade/pull/81

**Comment by tinywrkb** [comment]
How do you suggest handling file conflicts?

I think it would be better to have support for runtime/platform runner extensions.  
This could be used for Wine, Java, Mono, etc.  
Electron apps could optionally be built in a way that will run the Electron binary from the extension, similarly to what Arch Linux packages do.  
The app's maintainer could still have the possibility to deploy the runner from the sdk extension, and have it packaged with the app.  
The main blocker for implementing this is that we don't have support for automatically package a platform extension from a sdk extension unless we build a runtime and the sdk+platform extensions are split off the runtime.

**Comment by TheEvilSkeleton** [comment]
> How do you suggest handling file conflicts?

Sadly no. I haven't thought about the potential issues to be honest.

**Comment by RushingAlien** [comment]
> How do you suggest handling file conflicts?

I think handling it via prioritizing base apps that are first in order makes sense

**Comment by brunoais** [comment]
> First base app in the list will be prioritized

Agreed. It makes sense in many ways. That's also how [`overlayfs` works](https://www.kernel.org/doc/Documentation/filesystems/overlayfs.txt) so I'd use that to keep consistent.

> I think it should also use prioritization with the main app being prioritized the most?

Yes. I believe always. (If possible, an info or warning sharing that information would be useful)
